src/main.qml

```diff
@@ -56,60 +57,90 @@ Application {
     }
 
     Row {
-        anchors.verticalCenter: parent.verticalCenter
-        anchors.left: parent.left
-        anchors.right: parent.right
+        id: mainRow
+        anchors {
+            verticalCenter: parent.verticalCenter
+            verticalCenterOffset: -parent.height * 0.002
+            left: parent.left
+            leftMargin: parent.width * 0.159
```

Comment:
This is Dims.w(15.9)
Same for l67 which would be Dims.w(1.6)
It’s a shortcut for a screen size percentage. 

https://github.com/AsteroidOS/qml-asteroid/blob/master/src/controls/qml/Dims.qml#L28

Reply:
Yes, i informed myself about Dims when i first encountered them during looking into the HRM app and understand they are ment for adaption to non-square screens.
But i do not quite understand when to use them.
In the app drawers for instance i swapped Dims for parent.h/w * x since the Dims where not respected in the Appdrawer settings page.
Using Dims in WFs yields the same problem with the watchface settings page.
Is Dims only ment to be used in programs that are not wrapped/loaded in other qml files?
Or whats the rule here?

And it would be very kind if you could explain what is the difference to use root/parent/app.height * x strictly for vertical positioning and parent.width * x strictly for horizontal positioning and the Dims concept?
Afaiu just not mixing up height and width should also scale to any kind of screen ratio properly?
Is Dims here to prevent that mix up?

Haha, this piled up more questions then i liked. Sry, and thanks for your time!

Reply:
> Yes, i informed myself about Dims when i first encountered them during looking into the HRM app and understand they are ment for adaption to non-square screens.

Not necessarily. It's just a convenient way to express percentages of the screen width, percentages of the screen height, or percentages of the smallest dimension (for the very few waches that don't have a 1:1 aspect ratio and which we don't actually support yet anyway, or also for watches with a "flat tire", an unusable area at the bottom of a round screen)

This is more meant to be a shortcut to keep code a little bit nicer, ie:
`Dims.w(10)` instead of `app.width * 0.1` and it becomes really much nicer for text sizes where `Dims.l(7)` is more readable than app.width * 0.07 etc...

> But i do not quite understand when to use them.

When it makes the code look a bit nicer

>In the app drawers for instance i swapped Dims for parent.h/w * x since the Dims where not respected in the Appdrawer settings page. Using Dims in WFs yields the same problem with the watchface settings page. Is Dims only ment to be used in programs that are not wrapped/loaded in other qml files? Or whats the rule here?

I see, yeah this is very much a special case because the same QML code is meant to be loaded in two very different contexts (both asteroid-launcher and asteroid-settings), in one case it can assume it has the whole screen, in the other it can't. But for all other use cases than watchfaces and loaders, the apps know they always have full control of the screen and can safely use this.

> And it would be very kind if you could explain what is the difference to use root/parent/app.height * x strictly for vertical positioning and parent.width * x strictly for horizontal positioning and the Dims concept? Afaiu just not mixing up height and width should also scale to any kind of screen ratio properly? Is Dims here to prevent that mix up?

This is mostly for readability.

> Haha, this piled up more questions then i liked. Sry, and thanks for your time!

Np, you're welcome :)